pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/objects/channel/GetAllChannelMetadata.ios.kt

```diff
@@ -51,25 +50,27 @@ class GetAllChannelMetadataImpl(
             },
             includeCount = includeCount,
             includeCustom = includeCustom,
-            onSuccess = callback.onSuccessHandler {
-                PNChannelMetadataArrayResult(
-                    status = it?.status()?.toInt() ?: 0,
-                    data = (it?.data() as List<PubNubChannelMetadataObjC>).map { metadata ->
-                        PNChannelMetadata(
-                            id = metadata.id(),
-                            name = metadata.name(),
-                            description = metadata.descr(),
-                            custom = metadata.custom() as? Map<String, Any?>,
-                            updated = metadata.updated(),
-                            eTag = metadata.eTag(),
-                            type = metadata.type(),
-                            status = metadata.status()
-                        )
-                    },
-                    totalCount = it.totalCount()?.intValue ?: 0,
-                    next = it.next()?.end()?.let { hash -> PNPage.PNNext(pageHash = hash) },
-                    prev = it.next()?.start()?.let { hash -> PNPage.PNPrev(pageHash = hash) }
-                )
+            onSuccess = { data, totalCount, next ->
```

Comment:
The same, I also removed Objective-C class like `PubNubGetChannelMetadataResultObjC` and moved all parameters in the closure instead. Is it a better approach or not?

Reply:
Done. I did the same approach that was mentioned previously [here](https://github.com/pubnub/kotlin/pull/229#discussion_r1633431434)